lib/ansible/modules/cloud/vmware/vmware_guest_disk_controller.py

```diff
@@ -0,0 +1,488 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, Ansible Project
+# Copyright: (c) 2019, VMware, Inc. All Rights Reserved.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: vmware_guest_disk_controller
+short_description: Manage disk controllers related to virtual machine in given vCenter infrastructure
+description:
+    - This module can be used to add, remove disk controllers belonging to given virtual machine.
+    - All parameters and VMware object names are case sensitive.
+version_added: 2.9
+author:
+    - Diane Wang (@Tomorrow9) <dianew@vmware.com>
+notes:
+    - Tested on vSphere 6.0, 6.5 and 6.7
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+   name:
+     description:
+     - Name of the virtual machine.
+     - This is a required parameter, if parameter C(uuid) or C(moid) is not supplied.
+     type: str
+   uuid:
+     description:
+     - UUID of the instance to gather facts if known, this is VMware's unique identifier.
+     - This is a required parameter, if parameter C(name) or C(moid) is not supplied.
+     type: str
+   moid:
+     description:
+     - Managed Object ID of the instance to manage if known, this is a unique identifier only within a single vCenter instance.
+     - This is required if C(name) or C(uuid) is not supplied.
+     type: str
+     version_added: '2.9'
+   folder:
+     description:
+     - Destination folder, absolute or relative path to find an existing guest.
+     - This is a required parameter, only if multiple VMs are found with same name.
+     - The folder should include the datacenter. ESX's datacenter is ha-datacenter
+     - 'Examples:'
+     - '   folder: /ha-datacenter/vm'
+     - '   folder: ha-datacenter/vm'
+     - '   folder: /datacenter1/vm'
+     - '   folder: datacenter1/vm'
+     - '   folder: /datacenter1/vm/folder1'
+     - '   folder: datacenter1/vm/folder1'
+     - '   folder: /folder1/datacenter1/vm'
+     - '   folder: folder1/datacenter1/vm'
+     - '   folder: /folder1/datacenter1/vm/folder2'
+   datacenter:
+     description:
+     - The datacenter name to which virtual machine belongs to.
+     required: True
+   use_instance_uuid:
+     description:
+     - Whether to use the VMware instance UUID rather than the BIOS UUID.
+     default: no
+     type: bool
+     version_added: '2.8'
+   disk_controller:
+     description:
+     - A list of disk controllers to add or remove.
+     - The virtual disk controllers information is provided using this list.
+     - All values and parameters are case sensitive.
+     - 'Valid attributes are:'
```

Comment:
Can you please check if `suboption` can be used here - just like - suboptions in `vmware_dvs_host` module - https://github.com/ansible/ansible/blob/49da47d09fa72670d3f87bf16305d7cccd0465c1/lib/ansible/modules/cloud/vmware/vmware_dvs_host.py#L272

Reply:
Ok, I'll check how to use it and consider if I can use it here. Thanks.

Reply:
So the advantage of suboption is parameters's checking can be omitted writing in the module file?